File: backend-dotnet/src/BiteRight.Application/Queries/Categories/Search/SearchRequest.cs
using BiteRight.Application.Dtos.Common;
using MediatR;

namespace BiteRight.Application.Queries.Categories.Search;

public class SearchRequest : IRequest<SearchResponse>
{
    public string Query { get; set; }
    public PaginationParams PaginationParams { get; set; }

    public SearchRequest(
        string query,
        PaginationParams paginationParams
    )
    {
        Query = query;
        PaginationParams = paginationParams;
    }
}

File: mobile/app/src/main/java/com/sobczal2/biteright/data/api/requests/categories/SearchRequest.kt
package com.sobczal2.biteright.data.api.requests.categories

import com.google.gson.annotations.SerializedName
import com.sobczal2.biteright.dto.common.PaginationParams

data class SearchRequest(
    @SerializedName("query") val query: String,
    @SerializedName("paginationParams") val paginationParams: PaginationParams,
)